### PR TITLE
[Python][Text] Add methods to RexExpUtility class & StringUtility class

### PR DESCRIPTION
**Pull request by Aliandi**
This PR should be merged after # 1643 

### Changes made
- Changed files: `extractors.py` and `utilities.py`
- Add the method `index_of` to StringUtility and remove it from `extractors.py`
- Add the method `get_matches` to RexExpUtility and remove it from `extractors.py`

### CODE REVIEW

File: Python/libraries/recognizers-choice/recognizers_choice/choice/constants.py~HEAD [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Constants:
```
Comment (gasper-az) [inline Python/libraries/recognizers-choice/recognizers_choice/choice/constants.py~HEAD]: Should this file be included?

File: Python/libraries/recognizers-text/recognizers_text/parser.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 from typing import Optional
 from .extractor import ExtractResult
 
+
```
Comment (gasper-az) [inline Python/libraries/recognizers-text/recognizers_text/parser.py]: Can you remove this extra line??

File: Python/libraries/recognizers-text/recognizers_text/parser.py [hunk 3]
```diff
@@ -14,6 +15,7 @@ def __init__(self, source: ExtractResult = None):
             self.type = source.type
             self.data = source.data
 
+
```
Comment (gasper-az) [inline Python/libraries/recognizers-text/recognizers_text/parser.py]: And also this one?

File: Python/tests/test_runner_choice.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import pytest
+from runner import get_specs
+from recognizers_choice import recognize_boolean
+
+MODELFUNCTION = {'Boolean': recognize_boolean}
+
+
```
Comment (gasper-az) [inline Python/tests/test_runner_choice.py]: Can you remove this extra line?

File: Python/tests/test_runner_choice.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+@pytest.mark.parametrize('culture, model, options, context, source, expected_results', get_specs(recognizer='Choice', entity='Model'))
+def test_choice_recognizer(culture, model, options, context, source, expected_results):
+    results = get_results(culture, model, source)
+    assert len(results) == len(expected_results)
+    for actual, expected in zip(results, expected_results):
+        assert actual.type_name == expected['TypeName']
+        assert actual.text == expected['Text']
+        assert actual.resolution['value'] == expected['Resolution']['value']
+
+
```
Comment (gasper-az) [inline Python/tests/test_runner_choice.py]: Can you remove this extra line?

File: Python/tests/test_runner_number.py [hunk 6]
```diff
@@ -8,6 +8,7 @@
     'Percent': recognize_percentage,
 }
 
+
```
Comment (gasper-az) [inline Python/tests/test_runner_number.py]: Can you remove this extra line?

File: Python/libraries/recognizers-choice/setup.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+
+from setuptools import setup, find_packages
+
```
Comment (gasper-az) [inline Python/libraries/recognizers-choice/setup.py]: Can you remove this extra line?

File: Python/libraries/recognizers-choice/recognizers_choice/choice/english/boolean.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import regex
+from recognizers_choice.choice.extractors import BooleanExtractorConfiguration
+from recognizers_choice.resources.english_choice import EnglishChoice
+from recognizers_text.utilities import RegExpUtility
+from typing import Pattern
+
```
Comment (gasper-az) [inline Python/libraries/recognizers-choice/recognizers_choice/choice/english/boolean.py]: Can you remove this extra line?

File: Python/libraries/recognizers-choice/recognizers_choice/choice/recognizers_choice.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
+from enum import IntFlag
+from typing import List
+
+from recognizers_choice.choice.extractors import BooleanExtractor
+from recognizers_choice.choice.english import EnglishBooleanExtractorConfiguration
+from recognizers_choice.choice.models import BooleanModel
+from recognizers_choice.choice.parsers import BooleanParser
+from recognizers_text import Culture, Recognizer, ModelResult, Model
+
```
Comment (gasper-az) [inline Python/libraries/recognizers-choice/recognizers_choice/choice/recognizers_choice.py]: Can you remove this extra line?

File: Python/libraries/recognizers-choice/recognizers_choice/choice/recognizers_choice.py [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+from recognizers_choice.choice.extractors import BooleanExtractor
+from recognizers_choice.choice.english import EnglishBooleanExtractorConfiguration
+from recognizers_choice.choice.models import BooleanModel
+from recognizers_choice.choice.parsers import BooleanParser
+from recognizers_text import Culture, Recognizer, ModelResult, Model
+
+
+class ChoiceOptions(IntFlag):
+    NONE = 0
+
```
Comment (gasper-az) [inline Python/libraries/recognizers-choice/recognizers_choice/choice/recognizers_choice.py]: Can you remove this extra line?

File: Python/libraries/recognizers-choice/recognizers_choice/choice/recognizers_choice.py [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+def recognize_boolean(query: str,
+                      culture: str,
+                      options: ChoiceOptions = ChoiceOptions.NONE,
+                      fallback_to_default_culture: bool = True) -> List[ModelResult]:
+    recognizer = ChoiceRecognizer(culture, options)
+    model = recognizer.get_boolean_model(culture, fallback_to_default_culture)
+    return model.parse(query)
+
```
Comment (gasper-az) [inline Python/libraries/recognizers-choice/recognizers_choice/choice/recognizers_choice.py]: Can you remove this extra line?

File: Python/libraries/recognizers-choice/recognizers_choice/choice/parsers.py [hunk 12]
```diff
@@ -0,0 +1,66 @@
+from typing import Dict, List
+
+from recognizers_choice.choice.extractors import ChoiceExtractDataResult
+from recognizers_choice.choice.constants import Constants
+from recognizers_text.extractor import ExtractResult
+from recognizers_text.parser import Parser, ParseResult
+
```
Comment (gasper-az) [inline Python/libraries/recognizers-choice/recognizers_choice/choice/parsers.py]: Extra lines in this file

File: Python/libraries/recognizers-choice/recognizers_choice/choice/extractors.py [hunk 13]
```diff
@@ -0,0 +1,171 @@
+from abc import ABC
+from typing import List, Dict, Pattern
+
+from grapheme.api import slice
+import regex
+from recognizers_text import StringUtility, RegExpUtility
+from recognizers_text.extractor import Extractor, ExtractResult
+
+from .constants import *
+
```
Comment (gasper-az) [inline Python/libraries/recognizers-choice/recognizers_choice/choice/extractors.py]: Extra lines

**Review by gasper-az — CHANGES_REQUESTED** [review]
In general, LGTM. However, there are a few extra-lines that might be removed

File: Python/libraries/recognizers-text/recognizers_text/parser.py (hunk 2 above)
Comment (Aliandi) [reply Python/libraries/recognizers-text/recognizers_text/parser.py]: Its not an extra line! Python requires two newllines before a class

File: Python/tests/test_runner_choice.py (hunk 4 above)
Comment (Aliandi) [reply Python/tests/test_runner_choice.py]: The line is required by Python

File: Python/tests/test_runner_choice.py (hunk 5 above)
Comment (Aliandi) [reply Python/tests/test_runner_choice.py]: The line is required by Python

File: Python/tests/test_runner_number.py (hunk 6 above)
Comment (Aliandi) [reply Python/tests/test_runner_number.py]: The line is required by Python

File: Python/libraries/recognizers-choice/recognizers_choice/choice/constants.py~HEAD (hunk 1 above)
Comment (Aliandi) [reply Python/libraries/recognizers-choice/recognizers_choice/choice/constants.py~HEAD]: Nope, thx!

File: Python/libraries/recognizers-text/recognizers_text/parser.py (hunk 2 above)
Comment (Aliandi) [reply Python/libraries/recognizers-text/recognizers_text/parser.py]: The line is required by Python

File: Python/libraries/recognizers-text/recognizers_text/parser.py (hunk 3 above)
Comment (Aliandi) [reply Python/libraries/recognizers-text/recognizers_text/parser.py]: The line is required by Python

**Pull request closed without merging**